runtime/legion/legion_ops.cc

```diff
@@ -3813,21 +3883,44 @@ namespace Legion {
             runtime->forest->reduce_across(
                                   src_requirements[idx], dst_requirements[idx],
                                   src_targets, dst_targets, this, 
-                                  local_sync_precondition, predication_guard);
+                                  local_sync_precondition, predication_guard,
+                                  trace_info);
           Runtime::trigger_event(local_completion, across_done);
+          if (is_recording())
+          {
+#ifdef DEBUG_LEGION
+            assert(tpl != NULL && tpl->is_recording());
+#endif
+            tpl->record_trigger_event(local_completion, across_done);
+          }
         }
         // Apply our changes to the version states
         // Don't apply changes to the source if we have a composite instance
         // because it is unsound to mutate the region tree that way
         if (src_composite == -1)
           src_versions[idx].apply_mapping(map_applied_conditions);
         dst_versions[idx].apply_mapping(map_applied_conditions); 
+#ifdef DEBUG_LEGION
+      dump_physical_state(&src_requirements[idx], idx);
+      dump_physical_state(&dst_requirements[idx], idx + src_requirements.size());
```

Comment:
Do we really need to be dumping physical state in debug mode?

Reply:
Well, this isn't my code, but the code that existed in the snapshot I took. The dump_physical_state function returns immediately if the user did not request the tree dump. I guess an alternative is to hoist that condition to every place we call this function. If you prefer that way, I can make changes.